silver_payu/payment_processors.py

```diff
@@ -142,28 +134,43 @@ def _charge_transaction(self, transaction):
 
         payment = TokenPayment(payment_details, token)
 
-        # TODO: handle connection failing
-        result = payment.pay()
+        try:
+            result = payment.pay()
+        except Exception as error:
+            transaction.fail(fail_reason='Connectivity issues between server '
```

Comment:
We should not fail hard on connectivity issues. `execute_transactions` should/commands will retry on another run.